drivers/hmis/app/models/hmis/access_group.rb

```diff
@@ -46,6 +46,20 @@ class Hmis::AccessGroup < ApplicationRecord
     )
   end
 
+  scope :contains_with_inherited, ->(entity) do
+    case entity
+    when Hmis::Hud::Project
+      collection_ids = Hmis::Hud::Project.
+        joins(:group_viewable_entities).
+        where(id: entity.id).
+        pluck('hmis_group_viewable_entities.collection_id')
+      where(id: collection_ids)
+    else
+      # todo @martha
+      raise 'not implemented'
+    end
```

Comment:
```suggestion
  additional_entities = case entity
  when Hmis::Hud::Project
    {
      'Hmis::Hud::Organization' => entity.organization.id, 
      'GrdaWarehouse::DataSource' => entity.data_source.id,
    } 
  when Hmis::Hud::Organization
    {
      'GrdaWarehouse::DataSource' => entity.data_source.id,
    }
  when Hmis::DataSource
    {}
  else
    raise "Unsupported entity type: #{entity.class.name}"
  end
  a_t = Hmis::GroupViewableEntity.arel_table
  # Find the collections involving this entity directly
  query = a_t[:entity_type].eq(entity.class.name).and(a_t[:entity_id].eq(entity.id))
  # Add collections where this entity is a sub-entity of the assigned entity
  additional_entities.each do |klass, id|
    query = query.or(a_t[:entity_type].eq(klass).and(a_t[:entity_id].eq(id)))
  end
  # Fetch the collections that include this entity, either directly, or through inehritance
  collection_ids = Hmis::GroupViewableEntity.where(query).pluck(:collection_id)
  # Return a clean AccessGroup scope
  where(id: collection_ids)
```

Reply:
Oh!  Sorry, I'm not that familiar with the HMIS permission scheme, it was, at one point, very similar to the warehouse version, but seems to have matured.  Yes, `hmis_group_viewable_entity_projects` looks to include the same relationships I added.  Do you (or maybe @ttoomey) know how to take advantage of that table?

It looks like `Hmis::Hud::Project.joins(:group_viewable_entity_projects)` might be the ticket, but I'm definitely not following that completely.  Maybe you had the project version covered already.  The Data Sources are simple, but maybe we should build a new view for organizations?  (Or maybe we only need to determine which projects someone has access to, which might mean what you had was sufficient.)

Reply:
Hm, ok, yes I think you're right that `.joins(:group_viewable_entity_projects)` is the ticket, because of how this relationship is set up on the [Project model](https://github.com/greenriver/hmis-warehouse/blob/9755f5b77568378ef4f160713db8579848c4d63d/drivers/hmis/app/models/hmis/hud/project.rb#L46) with `has_many :group_viewable_entities, through: :group_viewable_entity_projects`.

I also discovered that it looks like there are already helpers for `includes_entities` with inheritance on the [GroupViewableEntity](https://github.com/greenriver/hmis-warehouse/blob/9755f5b77568378ef4f160713db8579848c4d63d/drivers/hmis/app/models/hmis/group_viewable_entity.rb#L53-L55), though they seemed to be a bit out-of-date and untested, so I fixed one or two bugs and added some spec tests to reassure us of the behavior.

Reply:
@ttoomey and/or @eanders (no rush) I've re-requested your review for this part specifically -- I'd be very glad for either one of you to give it a second pair of eyes before I merge, since I haven't done that much work on permissions yet.